api.js

```diff
@@ -214,6 +216,38 @@ exports.init = function (sbot, config) {
     rootMetaFeedLock.release(null, mf)
   }
 
+  function v1FindOrCreate(details, cb) {
+    if (!details.feetformat) details.feedformat = 'classic'
+
+    findOrCreate((err, rootFeed) => {
+      if (err) return cb(err)
+
+      const v1Details = { feedpurpose: 'v1', feedformat: BB1 }
+      const v1Visit = (feed) => feed.feedpurpose === v1Details.feedpurpose
+
+      findOrCreate(rootFeed, v1Visit, v1Details, (err, v1Feed) => {
+        if (err) return cb(err)
+
+        const shardDetails = {
+          feedpurpose: pickShard(rootFeed.keys.id, details.feedpurpose),
```

Comment:
:hot_pepper: the spec says

> When adding a new application-specific subfeed to the tree, we need to determine the parent shard based on a "name", which is any UTF-8 string that the application can choose freely, but it is RECOMMENDED that this string be unique to the use case

However I've just used the `feedpurpose` of the leaf subfeed. Is this wrong? Should the spec say you should use the `feedpurpose`, or should we have a new field on details like `feedidkeykey` (id and key are already taken :cry:)

Reply:
Ah, I see, the spec doesn't actually say what the `name` should be. I guess the `name` should be predictable for peers who want to replicate the leaf, because they need to calculate the shard nibble themselves.

I can't remember why we didn't just say that `name = feedpurpose` when drafting the spec. Do you remember @arj03 ?